src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String DELIMITER = "\\W";
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File checkFile = new File(fromFileName);
+        File newFile = new File(toFileName);
+        readFromFile(checkFile);
+        String[] allText = readFromFile(checkFile).split(DELIMITER);
+        int countSupply = getCount(allText, SUPPLY);
+        int countBuy = getCount(allText, BUY);
+        int result = countSupply - countBuy;
+        String report = generateReport(countSupply, countBuy, result);
+        writeReportToFile(newFile, report);
```

Comment:
try doing calculation operations in the generateReport method
It is better to rewrite like this:
```suggestion
        String data = readFromFile(fromFileName);
        String report = generateReport(data);
        writeReportToFile(toFileName, report);
```